### PR TITLE
[PVR] Add option to hide disabled timers in timer window (trac #15914).

### PR DESCRIPTION
**Pull request by ksooo**
Implements http://trac.kodi.tv/ticket/15914, requested several times in the forum during the last months.

@Jalle19, @xhaggi  mind taking a look.

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -3641,7 +3641,11 @@ msgctxt "#848"
 msgid "Are you sure you want to stop this recording?"
 msgstr ""
 
-#empty string with id 849
+#. Label of the option to hide disabled timers (in confluence sideblade) in timer window.
+#: skin.confluence
+msgctxt "#849"
+msgid "Hide disabled"
+msgstr ""
```
Comment (MartijnKaijser) [inline addons/resource.language.en_gb/resources/strings.po]: Please choose a way higher number as these suck for searching the code

File: xbmc/pvr/timers/PVRTimers.cpp [hunk 2]
```diff
@@ -450,7 +455,8 @@ bool CPVRTimers::GetSubDirectory(const CPVRTimersPath &path, CFileItemList &item
       if ((timer->m_bIsRadio == bRadio) &&
           (timer->m_iParentClientIndex != PVR_TIMER_NO_PARENT) &&
           (timer->m_iClientId == iClientId) &&
-          (timer->m_iParentClientIndex == iParentId))
+          (timer->m_iParentClientIndex == iParentId) &&
+          (!bHideDisabled || (timer->m_state != PVR_TIMER_STATE_DISABLED)))
```
Comment (Jalle19) [inline xbmc/pvr/timers/PVRTimers.cpp]: Could some of these currently duplicated checks be factored out to a separate method?

File: xbmc/pvr/windows/GUIWindowPVRTimers.cpp [hunk 3]
```diff
@@ -223,6 +224,13 @@ bool CGUIWindowPVRTimers::OnMessage(CGUIMessage &message)
         Update(GetDirectoryPath());
         bReturn = true;
       }
+      else if (message.GetSenderId() == CONTROL_BTNHIDEDISABLEDTIMERS)
+      {
+        CSettings::GetInstance().ToggleBool(CSettings::SETTING_PVRTIMERS_HIDEDISABLEDTIMERS);
```
Comment (Jalle19) [inline xbmc/pvr/windows/GUIWindowPVRTimers.cpp]: Does this mean the setting can be toggled both from the sidebar and the system settings. IMO either is fine but not both. Do we have any other settings that behave like this?

File: xbmc/pvr/timers/PVRTimers.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/pvr/timers/PVRTimers.cpp]: Really? Nah... too less duplicate code, imo. It's actually only <code>timer->m_bIsRadio == bRadio</code> and <code>!bHideDisabled || (timer->m_state != PVR_TIMER_STATE_DISABLED)</code>. What to make out of this? And how to name the new method doing just these two checks??

File: xbmc/pvr/windows/GUIWindowPVRTimers.cpp (hunk 3 above)
Comment (ksooo) [inline xbmc/pvr/windows/GUIWindowPVRTimers.cpp]: Only from the sidebar, as settings level is 4. => https://github.com/xbmc/xbmc/pull/8543/files#diff-92b91a5c902936f6e9a03f26b054d7d6R1637

The settings in addon manager confluence sidebar (like "Hide incompatible") behave exactly like this. I actually "borrowed" the implementation ghist from there.

Comment (Jalle19) [inline xbmc/pvr/windows/GUIWindowPVRTimers.cpp]: Okay

File: xbmc/pvr/timers/PVRTimers.cpp (hunk 2 above)
Comment (Jalle19) [inline xbmc/pvr/timers/PVRTimers.cpp]: No problem, let's leave it like this.

**Comment by opdenkamp** [comment]
just my 2c: the rule is "every setting you add kills a nice fluffy bunny". for this, I would have done something like sorting the list by moving disabled entries to the bottom and have some in the skin to show that they've been disabled.

**Comment by ksooo** [comment]
@opdenkamp: I personally agree, but your approach seems to be to complicated for avarage Joe - at least for this particular use case. ;-)

**Comment by ksooo** [comment]
Regarding sorting: Is it possible to "define" a primary and a secondary sort criteria - for instance primary: state, secondary: start time?

**Comment by opdenkamp** [comment]
Would it? If you just put them in the bottom of the list and grey the entries out in the skin, or something else that makes it clear that the entries have been disabled, then I think that it would be less complicating. A setting, hidden in level 4, is harder to figure out for Average Joe than this (greyed out means disabled in most applications).

Anyway, like I said, just my 2c cause this PR happened to be on top this morning when I checked Github ;-)

**Comment by opdenkamp** [comment]
| Regarding sorting: Is it possible to "define" a primary and a secondary sort criteria - for instance primary: state, secondary: start time?

You can define them in the ViewState, but only some types have been implemented afaik (standard things like filename, size, etc.)

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**